synapse/lib/iq.py

```diff
@@ -689,6 +748,100 @@ def getDmonCore(self):
+            ('auth:add:rrule', {'role': 'deleter',
+                                'rule': ('node:del',
+                                         {'form': '*'})
+                                }),
+            ('auth:add:rrule', {'role': 'deleter',
+                                'rule': ('node:tag:del',
+                                         {'tag': '*'})
+                                }),
+            ('auth:add:rrule', {'role': 'deleter',
+                                'rule': ('node:prop:set',
+                                         {'form': '*', 'prop': '*'})
+                                }),
+            ('auth:add:urole', {'user': 'user@localhost', 'role': 'creator'}),
+            ('auth:add:urole', {'user': 'user@localhost', 'role': 'deleter'}),
+        )
+
+        with self.getDirCore(conf=conf) as core:
+            s_scope.set('syn:core', core)
+            dirn = s_scope.get('dirn')
+            writeCerts(dirn)
+            cafile = os.path.join(dirn, 'ca.crt')
+            keyfile = os.path.join(dirn, 'server.key')
+            certfile = os.path.join(dirn, 'server.crt')
+            userkey = os.path.join(dirn, 'user.key')
+            usercrt = os.path.join(dirn, 'user.crt')
+            rootkey = os.path.join(dirn, 'root.key')
+            rootcrt = os.path.join(dirn, 'root.crt')
+            with s_daemon.Daemon() as dmon:
+                s_scope.set('syn:dmon', dmon)
+                dmon.share('core', core)
+                link = dmon.listen('ssl://localhost:0/',
+                                   cafile=cafile,
+                                   keyfile=keyfile,
+                                   certfile=certfile,
+                                   )
+                s_scope.set('syn:test:link', link)
+                port = link[1].get('port')
+                url = 'ssl://user@localhost/core'
+                user_prox = s_telepath.openurl(url,
+                                               port=port,
+                                               cafile=cafile,
+                                               keyfile=userkey,
+                                               certfile=usercrt
+                                               )  # type: s_cores_common.CoreApi
+                root_prox = s_telepath.openurl(url,
+                                               port=port,
+                                               cafile=cafile,
+                                               keyfile=rootkey,
+                                               certfile=rootcrt
+                                               )  # type: s_cores_common.CoreApi
+
+                if configure_roles:
+                    for mesg in amesgs:
+                        isok, retn = root_prox.authReact(mesg)
+                        s_common.reqok(isok, retn)
+
+                try:
+                    yield user_prox, root_prox
+                finally:
+                    user_prox.fini()
```

Comment:
Do you need to do any of the `s_scope` stuff like above?

Reply:
I don't think so - the `dirn`, `core`, `dmon` and `link` are all in scope already.

Reply:
ok